### ISSUE TITLE
404 page not found

### ISSUE THREAD

**Issue by myde2001** [cylonchau/firewalld-gateway#6]
Hi, I can't get to install the app, after cloning the repository and building it with make, all I get after executing the binary and going to the URL of the web server is "404 page not found". Any idea of what's happening? Thanks

**Comment by cylonchau** [comment]
Hi bro, you can refer API document 
https://documenter.getpostman.com/view/12796679/UV5agGNr

**Comment by myde2001** [comment]
> Hi bro, you can refer API document https://documenter.getpostman.com/view/12796679/UV5agGNr

Hi, it doesn't work, I cannot find a way to get the WebUI working.

**Comment by cylonchau** [comment]
> > Hi bro, you can refer API document https://documenter.getpostman.com/view/12796679/UV5agGNr
> 
> Hi, it doesn't work, I cannot find a way to get the WebUI working.

You can try run in docker, if you want runinng in linux vm, you need install a nginx or other web server

**Comment by joe-at-startupmedia** [comment]
The static assets aren't served by the application. That's why an nginx configuration exists which listens on a different port which can be changed to anything (e.g. 443) https://github.com/cylonchau/firewalld-gateway/blob/aff50c483884cbb78ba1cc2ecb592669f8f17139/uranus.nginx.conf

**Comment by cylonchau** [comment]
will update

Best regards
________________________________
寄件者: joe-at-startupmedia ***@***.***>
寄件日期: 2024年8月20日 08:57
收件者: cylonchau/firewalld-gateway ***@***.***>
副本: Cylon ***@***.***>; Comment ***@***.***>
主旨: Re: [cylonchau/firewalld-gateway] 404 page not found (Issue #6)


The static assets aren't served by the application. That's why an nginx configuration exists which listens on a different port which can be changed to anything (e.g. 443) https://github.com/cylonchau/firewalld-gateway/blob/aff50c483884cbb78ba1cc2ecb592669f8f17139/uranus.nginx.conf

**Comment by cylonchau** [comment]
> The static assets aren't served by the application. That's why an nginx configuration exists which listens on a different port which can be changed to anything (e.g. 443) https://github.com/cylonchau/firewalld-gateway/blob/aff50c483884cbb78ba1cc2ecb592669f8f17139/uranus.nginx.conf

Please refer [install manual](https://www.oomkill.com/2024/08/uranus-installation/)

**Comment by cylonchau** [comment]
> Please refer [install manual](https://www.oomkill.com/2024/08/uranus-installation/)

**Comment by cylonchau** [comment]
> > Hi bro, you can refer API document https://documenter.getpostman.com/view/12796679/UV5agGNr
> 
> Hi, it doesn't work, I cannot find a way to get the WebUI working.

Please refer [install manual](https://www.oomkill.com/2024/08/uranus-installation/)

**Comment by joe-at-startupmedia** [comment]
@cylonchau no worries I was able to get the web assets working, I was trying to explain to the OP why the binary doesn't serve the static assets. Personally I had problems establishing communications with DBUS because the latest version doesn't allow tcp communication using the method explained.

**Comment by cylonchau** [comment]
> @cylonchau no worries I was able to get the web assets working, I was trying to explain to the OP why the binary doesn't serve the static assets. Personally I had problems establishing communications with DBUS because the latest version doesn't allow tcp communication using the method explained. I would open a separate issue but I have since moved on.

I haven't figured out how to embed static files into the binary yet. 😥, TCP need configure dbus enable allow remote connection, default is not allow.

**Comment by joe-at-startupmedia** [comment]
[cylonchau](https://github.com/cylonchau) give me a few, I'll open another issue explaining the TCP dbus issue.

**Comment by joe-at-startupmedia** [comment]
@cylonchau I'm getting a 404 trying to register a user:

Started process:
```
I0824 09:45:11.755718   17525 server.go:79] FLAG: --one_output="false"
I0824 09:45:11.755727   17525 server.go:79] FLAG: --skip_headers="false"
I0824 09:45:11.755743   17525 server.go:79] FLAG: --skip_log_headers="false"
I0824 09:45:11.755751   17525 server.go:79] FLAG: --sql-driver="sqlite"
I0824 09:45:11.755761   17525 server.go:79] FLAG: --stderrthreshold="2"
I0824 09:45:11.755771   17525 server.go:79] FLAG: --v="5"
I0824 09:45:11.755778   17525 server.go:79] FLAG: --vmodule=""
I0824 09:45:11.757932   17525 model.go:35] Databases stats is {MaxOpenConnections:100 OpenConnections:1 InUse:0 Idle:1 WaitCount:0 WaitDuration:0s MaxIdleClosed:0 MaxIdleTimeClosed:0 MaxLifetimeClosed:0}
I0824 09:45:11.758777   17525 http.go:27] Listening and serving HTTP on 0.0.0.0:2952
I0824 09:45:11.760529   17525 processor.go:60] Async event processor started, waitting task...
```

```
[root@startup-nginx-1 ~]# curl --location 'localhost:2952/ping'
{"code":10000,"msg":"operation succeeded","data":"pong"}
[root@startup-nginx-1 ~]# curl --location 'http://0.0.0.0:2952/auth/signup' --data '{"username": "admin","password": "111"}'
```

Any ideas?

**Comment by joe-at-startupmedia** [comment]
nvm, I figured it out: https://github.com/cylonchau/firewalld-gateway/blob/67b2e8092059a545ac5c9e32897ba096e8742434/docs/swagger.yaml#L1924

I was referring to that old postman docs. 

updated command to crate a user is: 
```bash
curl -X POST "http://0.0.0.0:2952/sso/signup" --data '{"username": "admin1","password": "1112"}'
```

**Comment by cylonchau** [comment]
> nvm, I figured it out:
> 
> https://github.com/cylonchau/firewalld-gateway/blob/67b2e8092059a545ac5c9e32897ba096e8742434/docs/swagger.yaml#L1924
> 
> I was referring to that old postman docs.
> 
> updated command to crate a user is:
> 
> ```shell
> curl -X POST "http://0.0.0.0:2952/sso/signup" --data '{"username": "admin1","password": "1112"}'
> ```

This feature must be enable Database config, if not, those router will not reigster, this is only a firewalld rest api.

Please refer [install manual](https://www.oomkill.com/2024/08/uranus-installation/), you must change config and migarte database.

```go
	if !config.CONFIG.MySQL.IsEmpty() || !config.CONFIG.SQLite.IsEmpty() {
		ssoRouter := &sso.SSO{}
		ssoRouter.RegisterUserAPI(ssoGroup)
		// auth route
		authRouter := &auth.Auth{}
		authRouter.RegisterUserAPI(authAPI)

		usersRouter := &user.User{}
		usersRouter.RegisterUserAPI(userAPI)

		tokenRouter := &Token.Token{}
		tokenRouter.RegisterTokenAPI(tokenAPI)

		tagRouter := &tag.Tag{}
		tagRouter.RegisterTagAPI(tagGroup)

		hostRouter := &host.Host{}
		hostRouter.RegisterHostAPI(hostGroup)

		asyncHostRouter := &host.AsyncHost{}
		asyncHostRouter.RegisterAsyncHostAPI(hostGroup)

		templateRouter := &template.Template{}
		templateRouter.RegisterTemplateAPI(templateGroup)

		auditRouter := &audit.Audit{}
		auditRouter.RegisterAuditAPI(auditAPI)
	}
}
```

**Comment by joe-at-startupmedia** [comment]
@cylonchau I already ran with `--migration`. This postman link is outdated: https://documenter.getpostman.com/view/12796679/UV5agGNr#060f80aa-7574-4dd1-b145-36847c784cbd

**Comment by cylonchau** [comment]
> @cylonchau I already ran with `--migration`. This postman link is outdated: https://documenter.getpostman.com/view/12796679/UV5agGNr#060f80aa-7574-4dd1-b145-36847c784cbd

yes, API doc is already build in, you can use /swagger/index.html API document.

**Comment by cylonchau** [comment]
> @cylonchau no worries I was able to get the web assets working, I was trying to explain to the OP why the binary doesn't serve the static assets. Personally I had problems establishing communications with DBUS because the latest version doesn't allow tcp communication using the method explained. I would open a separate issue but I have since moved on.

Latest version is can embed assets in binary file.
